stonesoup/resampler/particle.py

```diff
@@ -24,6 +24,7 @@ def resample(self, particles):
 
         n_particles = len(particles)
         weight = Probability(1/n_particles)
+        particles = sorted(particles, key=lambda x: x.weight, reverse=False)
```

Comment:
It would be better to use [`attrgetter`](https://docs.python.org/3/library/operator.html#operator.attrgetter) for the key.